### ISSUE TITLE
C:\\MinGw\\bin\\gcc.exe failed with exit status 1

### ISSUE THREAD

**Issue by Link-djst** [tgalal/yowsup#1468]
I´m trying to use yowsup, but i can´t install this. As the title, that´s the error i found everytime i try to install it. C:\MinGw\bin\gcc.exe failed with exit status 1. I have the path of minGW C:\MinGW\bin and i have already installed gcc from the command line with mingw-get.exe install gcc

**Comment by freelo** [comment]
gcc usually gives you more than only its exit status. Try to get a look at its output.

**Comment by Link-djst** [comment]
@freelo Before that shows this message 
''C:\Python27\libs/libpython27.a error adding symbols: File format not recognized

collect2.exe: error: ld return 1 exit state]
error: command: C:\\MinGw\\bin\\gcc.exe failed with exit status 1'

**Comment by freelo** [comment]
Where does that libpython27.a come from? Did you compile it yourself? By the way, you did not say what command you were executing to get that error (and the steps before).

**Comment by Link-djst** [comment]
1. I installed minGW and then i added to the path.
2. then I created the PYTHONPATH\Lib\distutils\distutils.cfg with '[build]
compiler=mingw32'
3. after that i had to install 'mingw-get.exe install gcc' in my command line
4. 'mingw-get.exe install zlib' in my command line.
5. python setup.py install in my command line from the yowsup-master folder. 

that´s all i have done so far... 

My pythonxy is 2.7 and i have windows 10 64 bits. That libpython27.a i did not created it.

**Comment by freelo** [comment]
Thanks for the info. It's just as the instructions say (but they don't mention how to install python, that's strange).
Could you do an `objdump -t libpython27.a` to check its contents? And since you are using mingw32, please check that you have installed 32bit python as well.

**Comment by Link-djst** [comment]
Yes, I have python 32bits with mingw32. So.. you dont have that file, maybe im going to uninstall python xy. 

This is the output of `objdump -t libpython27.a`

SYMBOL TABLE:
[  0](sec  1)(fl 0x00)(ty   0)(scl   3) (nx 0) 0x00000000 .text
[  1](sec  2)(fl 0x00)(ty   0)(scl   3) (nx 0) 0x00000000 .data
[  2](sec  3)(fl 0x00)(ty   0)(scl   3) (nx 0) 0x00000000 .bss
[  3](sec  4)(fl 0x00)(ty   0)(scl   3) (nx 0) 0x00000000 .idata$7
[  4](sec  5)(fl 0x00)(ty   0)(scl   3) (nx 0) 0x00000000 .idata$5
[  5](sec  6)(fl 0x00)(ty   0)(scl   3) (nx 0) 0x00000000 .idata$4
[  6](sec  7)(fl 0x00)(ty   0)(scl   3) (nx 0) 0x00000000 .idata$6
[  7](sec  1)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x00000000 __PyUnicodeUCS2_IsNumeric
[  8](sec  5)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x00000000 __imp___PyUnicodeUCS2_IsNumeric
[  9](sec  0)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x00000000 __head_C__build27_cpython_PCBuild_libpython27_a



dmmes00997.o:     file format pe-i386

and there are more tables like this

**Comment by freelo** [comment]
Yes, I am not using that version(msi crashes anyways), but I have seen the file in the MSI package. Your reported file format `format pe-i386` seems right as well. Can you identify the last command executed? It is collect2 that is failing, but what was the gcc command that led to it?

I'm just guessing here, since, on windows, I just installed cygwin's python/gcc, then setuptools and pip.

**Comment by marius00** [comment]
If you're on windows, use Py 2.7.9 or lower.

The curve lib fails to compile with version >= Py 2.7.10 on windows.

**Comment by manmanMM** [comment]
So if in  this way, is there any method to make it succeed with like Py 2.7.12? Thanks !

**Comment by pablo-dave** [comment]
Hi, can you help me with this error please: 
runing: python 2.7.8 64 bit, cygwin32

collect2.exe: error: ld devolvió el estado de salida 1
error: Setup script exited with error: command 'C:\\MinGW\\bin\\gcc.exe' failed with exit status 1

**Comment by freddybarco** [comment]
''C:\Python27\libs/libpython27.a error adding symbols: File format not recognized

collect2.exe: error: ld return 1 exit state]
error: command: C:\MinGw\bin\gcc.exe failed with exit status 1'

[Solucionado]
Es increíble , pero lo unico que tenia que hacer fue eliminar temporalmente dicho archivo "libpython27.a"
y volver a correr la instalación. Y con eso se soluciono .

**Comment by freddybarco** [comment]
''C:\Python27\libs/libpython27.a error adding symbols: File format not recognized

collect2.exe: error: ld return 1 exit state]
error: command: C:\MinGw\bin\gcc.exe failed with exit status 1'

[Solucionado]
Lo único que tienen que hacer es eliminar temporalmente el archivo "libpython27.a" .Y volver a correr la instalación.

**Comment by kevinkurek** [comment]
Ever get a conclusive solution here? Created Python 2.7.9 virtualenv and get the same error. Python 3.x compiles fine.

**Comment by karimQasem** [comment]
same error help please

**Comment by bernardob8cargo** [comment]
same error here... 

**Failed building wheel for python-axolotl-curve25519**
Running setup.py clean for python-axolotl-curve25519
Failed to build python-axolotl-curve25519
jsonschema 3.0.1 has requirement six>=1.11.0, but you'll have six 1.10.0 which is incompatible.

right after
---------------------------------------------
Command "C:\Users\X\Anaconda3\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\X\\AppData\\Local\\Temp\\pip-install-wuevpz96\\python-axolotl-curve25519\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\X\AppData\Local\Temp\pip-record-q2hhjcoc\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\X\AppData\Local\Temp\pip-install-wuevpz96\python-axolotl-curve25519\

I do have six updated!
---------------------------------
(base) C:\Users\X>pip install --upgrade six 1.11.0
**Requirement already up-to-date: six in c:\users\X\anaconda3\lib\site-packages (1.12.0)**
Collecting 1.11.0
  Could not find a version that satisfies the requirement 1.11.0 (from versions: )
No matching distribution found for 1.11.0
